src/jdk.compiler/share/classes/com/sun/tools/javac/comp/LambdaToMethod.java

```diff
@@ -1152,7 +1152,7 @@ VarSymbol translate(final VarSymbol sym, LambdaSymbolKind skind) {
                     propagateAnnos = false;
                     break;
                 case LOCAL_VAR:
-                    ret = new VarSymbol(sym.flags() & FINAL, sym.name, sym.type, translatedSym);
+                    ret = new VarSymbol(sym.flags() & (SYNTHETIC | FINAL), sym.name, sym.type, translatedSym);
```

Comment:
this will effectively make synthetic all local variables, should we do this starting from source 27?, what about the parameter being generated below? should it be synthetic too?

Reply:
Exactly - this code should copy the specified flags, if they are set. I.e. if the original variable is not synthetic, the new one won't be synthetic either. If the original variable is synthetic, the flag will be transferred to the new variable.

Given the handling of local variables and parameters is slightly different, I am not sure unifying the flags manipulation is quite feasible.